server/pixabay_utils.py

```diff
@@ -39,14 +40,16 @@ def fetch_images_tag(pixabay_search_keyword, num_images):
     output = response.json()
 
     all_images = []
+    all_image_urls = []
     start_time = time.time()
     for each in output["hits"]:
         imageurl = each["webformatURL"]
         response = requests.get(imageurl)
         image = Image.open(BytesIO(response.content)).convert("RGB")
         all_images.append(image)
+        all_image_urls.append(imageurl)
 
     end_time = time.time() - start_time
     logging.info(f"Fetched individual results in {end_time:.3f} " f"seconds.")
 
-    return all_images
+    return all_images, all_image_urls
```

Comment:
Why return images at all in that case?

Reply:
This is the same reason that we have to feed images into the model, so we need both raw image data and the urls